File: Programmers/로또의 최고 순위와 최저 순위.py
"""
def solution(lottos, win_nums):
    ans_min = []
    for win_num in win_nums:
        if win_num in lottos:
            ans_min.append(win_num)
    ans_max = lottos.count(0) + len(ans_min)
    if len(ans_min) == 0:
        ans_min.append(0)
    if ans_max == 0:
        ans_max = 1
    answer = [7-ans_max, 7-len(ans_min)]
    return answer
"""
def solution(lottos, win_nums):
    rank = [1,2,3,4,5,6,6]
    min_num = len(set(win_nums) - set(lottos))
    max_num = min_num - lottos.count(0)
    return rank[max_num], rank[min_num]

File: Programmers-JAVA/로또의 최고 순위와 최저 순위.java
import java.util.*;
import java.lang.*;

class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = {};
        int zero = 0;
        int match = 0;
        Arrays.sort(lottos);
        Arrays.sort(win_nums);
        
        for(int lotto: lottos) {
            if(lotto == 0) zero++;
            else {
                for(int win_num: win_nums) {
                    if(lotto == win_num)
                        match++;
                }
            }
        }
        answer = new int[2];
        answer[0] = (zero+match)==0?6:7-(zero+match);
        answer[1] = match==0?6:7-match;
        
        System.out.println(zero+", "+match);
        return answer;
    }
}
